packages/lib/src/acting-io.ts

```diff
@@ -11,29 +11,32 @@ import { SmartThingsCommandInterface } from './smartthings-command'
 
 // TODO: implement equivalent of acceptIndexId from old code
 
-export type SelectingCommand<L> = SmartThingsCommandInterface & Sorting & CommonListOutputProducer<L>
+export type SelectingConfig<L> = Sorting & Naming & CommonListOutputProducer<L>
 
-export async function selectGeneric<ID, L>(command: SelectingCommand<L>, id: ID | undefined,
-		listItems: ListDataFunction<L>, getIdFromUser: IdRetrievalFunction<ID, L>): Promise<ID> {
-	if (id) {
-		return id
+export async function selectGeneric<ID, L>(command: SmartThingsCommandInterface, config: SelectingConfig<L>,
+		preselectedId: ID | undefined, listItems: ListDataFunction<L>,
+		getIdFromUser: IdRetrievalFunction<ID, L>, promptMessage?: string): Promise<ID> {
+	if (preselectedId) {
+		return preselectedId
 	}
 
-	const list = await outputList(command, listItems, true)
+	const list = await outputList(command, config, listItems, true)
 	if (list.length === 0) {
 		// Nothing was found; user was already notified.
 		command.exit(0)
 	}
 
-	return await getIdFromUser(command, list)
+	return await getIdFromUser(config, list, promptMessage)
 }
 
-export async function select<L>(command: SelectingCommand<L>, id: string | undefined,
-		listItems: ListDataFunction<L>): Promise<string> {
-	return selectGeneric(command, id, listItems, stringGetIdFromUser)
+export async function selectFromList<L>(command: SmartThingsCommandInterface, config: SelectingConfig<L>,
+		preselectedId: string | undefined, listItems: ListDataFunction<L>, promptMessage?: string): Promise<string> {
```

Comment:
`promptMessage` unused?

Reply:
Oops...was supposed to be passed on.

Reply:
Added a unit test to cover this case too now.